### PR TITLE
Condense output in presence of multiple suggestions

### PR DESCRIPTION
**Pull request by oli-obk**
part of #41

### CODE REVIEW

File: tests/tests/libui-rs/simple/output.txt [hunk 1]
```diff
@@ -38,11 +36,11 @@ try
 [1;34m  -->[0m src/controls.rs:373:5-378:6
 
 try this
-[0]: [1;33mSuggestion - Replace:[0m
+[0]: [1;33mAt:[0m
+[31m⍉[0mpub fn new() -> Entry {
```
Comment (oli-obk) [inline tests/tests/libui-rs/simple/output.txt]: Fun bug, but this is a clippy bug, so I won't fix it here. Clippy should be placing this suggestion properly.

File: src/main.rs [hunk 2]
```diff
@@ -164,19 +193,23 @@ fn handle_suggestions(
         let mut i = 0;
         for solution in suggestion.solutions.iter() {
             println!("\n{}", solution.message);
+
+            // check whether we can squash all suggestions into a list
+            if solution.replacements.len() > 1 {
+                let first = solution.replacements[0].clone();
+                if solution.replacements.iter().all(|s| first.snippet.file_name == s.snippet.file_name && first.snippet.line_range == s.snippet.line_range) {
```
Comment (killercup) [inline src/main.rs]: Can you put this in a `let all_suggestion_are_alternatives_to_each_other = …` (or something shorter)? Then, this line becomes the much more readable `if all_suggestion_are_alternatives_to_each_other {`.

File: src/main.rs [hunk 3]
```diff
@@ -164,19 +193,23 @@ fn handle_suggestions(
...
+                    prelude(&first);
+                    for suggestion in solution.replacements.iter() {
```
Comment (killercup) [inline src/main.rs]: Didn't clippy suggest `&solution.replacements` here?

File: src/main.rs [hunk 4]
```diff
@@ -139,6 +139,35 @@ enum AutofixMode {
...
+    let snippet = &suggestion.snippet;
+    if snippet.text.1.is_empty() {
+        if suggestion.replacement.ends_with('\n') || suggestion.replacement.starts_with('\n') {
+            println!("{}", "Insert line:".yellow().bold());
+        } else {
+            println!("{}", "At:".yellow().bold());
+            println!(
+                "{lead}{text}{tail}",
+                lead = indent(4, &snippet.text.0),
+                text = "⍉".red(),
```
Comment (killercup) [inline src/main.rs]: What's up with the [apl functional symbol circle backslash](http://graphemica.com/⍉) here? Do you mean to convey that the text is empty here? Then: Did you mean to use the [empty set](http://graphemica.com/∅)—or actually [ε](http://graphemica.com/ε)?

File: src/main.rs [hunk 5]
```diff
@@ -139,6 +139,35 @@ enum AutofixMode {
     Yolo,
 }
 
+fn prelude(suggestion: &Replacement) {
+    let snippet = &suggestion.snippet;
+    if snippet.text.1.is_empty() {
+        if suggestion.replacement.ends_with('\n') || suggestion.replacement.starts_with('\n') {
```
Comment (killercup) [inline src/main.rs]: What are you testing here? What meaning does a replacement starting or ending with a newline have? Can you maybe rewrite this as `let meaningful_name_of_usecase_here = …; if meaningful_name_of_usecase_here {`?

File: src/main.rs [hunk 6]
```diff
@@ -139,6 +139,35 @@ enum AutofixMode {
...
+            println!("{}", "Insert line:".yellow().bold());
+        } else {
+            println!("{}", "At:".yellow().bold());
+            println!(
+                "{lead}{text}{tail}",
+                lead = indent(4, &snippet.text.0),
```
Comment (killercup) [inline src/main.rs]: Memo for later refactoring: Turn `text` from `(String, String, String)` into `struct ReplacementText { leading: String, content: String, trailing: String, }`

File: src/main.rs (hunk 5 above)
Comment (oli-obk) [reply src/main.rs]: öhm... the use case is suggestions which wanna be on their own line... by being inserted after or before another line.

File: src/main.rs (hunk 4 above)
Comment (oli-obk) [reply src/main.rs]: I just wanted a pretty symbol. I would have preferred something like the google maps "you are here" marker. You know. with the pointy bottom and the round top.

File: src/main.rs (hunk 3 above)
Comment (oli-obk) [reply src/main.rs]: I think clippy throws up all kinds of things when used on rustfix. I never used it ;) I should. And I should feel bad for not having it done yet. ... Ah now I remember I wanted to fix rustfix with rustfix, but some clippy suggestions were breaking it. So I wanted to fix those first. I'm probably trapped in a loop

File: src/main.rs (hunk 4 above)
Comment (killercup) [reply src/main.rs]: like, ▲ or ▼?

File: src/main.rs (hunk 5 above)
Comment (killercup) [reply src/main.rs]: Doh! Okay, that is pretty obvious…

-ly to be named `let wants_to_be_on_own_line` :D

File: src/main.rs (hunk 4 above)
Comment (oli-obk) [reply src/main.rs]: idc. If you have sth you like, I'll use it.

Comment (killercup) [reply src/main.rs]: It's probably best to use something people actually have in their terminal font… so, v or ^? I don't mind either way

Comment (killercup) [reply src/main.rs]: if you want to change it, go ahead, otherwise i'm fine with merging with backslashed circle.

Comment (oli-obk) [reply src/main.rs]: I'll change it when I get back to a PC. Github web editing on mobile is a pain and would produce many commits ;)

**Comment by killercup** [comment]
Thanks!

bors r+

**Pull request merged**